app/uk/gov/hmrc/gform/controllers/ErrResponder.scala

```diff
@@ -62,23 +62,31 @@ class ErrResponder(
   def forbidden(
     message: String,
     messageHtml: Option[Html] = None,
-    availableLanguages: Option[AvailableLanguages] = None
+    availableLanguages: Option[AvailableLanguages] = None,
+    isUpstreamError: Boolean = false
   )(implicit request: RequestHeader): Future[Result] =
-    forbiddenReason(message, "generic.error.pageRestricted", messageHtml, availableLanguages)
+    forbiddenReason(message, "generic.error.pageRestricted", isUpstreamError, messageHtml, availableLanguages)
 
-  def forbiddenWithReason(reason: String)(implicit request: RequestHeader): Future[Result] =
-    forbiddenReason(reason, reason)
+  def forbiddenWithReason(reason: String, isUpstreamError: Boolean = false)(implicit
+    request: RequestHeader
+  ): Future[Result] =
+    forbiddenReason(reason, reason, isUpstreamError)
 
   private def forbiddenReason(
     message: String,
     reason: String,
+    isUpstreamError: Boolean,
     reasonHtml: Option[Html] = None,
     availableLanguages: Option[AvailableLanguages] = None
   )(implicit
     request: RequestHeader,
     messages: Messages
   ): Future[Result] = {
-    logger.info(s"Trying to access forbidden resource: $message")
+    if (isUpstreamError)
+      logger.error(s"Trying to access forbidden resource: $message")
+    else
+      logger.info(s"Trying to access forbidden resource: $message")
```

Comment:
To reduce duplications, can we refactor this logging logic into a class, something like:

```
class private CleverLogger(isUpstreamError: Boolean) {
  def log(msg: String) = if (isUpstreamError)
    logger.error(msg)
  else
    logger.info(msg)
  }
  
object CleverLogger {
    // constructors
    val LocalLogger = new CleverLogger(false)
    val UpstreamLogger = new CleverLogger(true)
}
```

and to pass `CleverLogger` around instead of `isUpstreamError: Boolean = false`

Reply:
Thanks @josef-vlach 
I have made changes as per suggestions. I have used the default parameter because defs `forbidden` and `forbiddenWithReason` in `ErrResponder` have been called from `AuthenticatedRequestActions` other than `ErrorHandler`.